desktop/bin/make-changelog.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+CALYPSO_DIR=$(cd $(dirname $0)/../../ && pwd)
+
+# desktop tags
+function get_desktop_tags() {
+  desktop_tags=""
+  declare -i x=0
+  for tag in $(git tag); do
+    # guard against stray v9.9.9 test tag that keeps being re-introduced to the repo! ಠ_ಠ
+    if [[ "$tag" == *"9.9.9" ]]; then
+      continue;
+    fi
+
+    if [[ $tag == v* ]]; then
+      if [ $x -gt 0 ]; then
+        desktop_tags="$desktop_tags"$'\n'"$tag"
+      else
+        desktop_tags="$tag"
+      fi
+    fi
+    x+=1
+  done
+  echo "$desktop_tags"
+}
+
+# fetch all tags in descending lexicographical order
+# (exclude current tag with `awk`)
+tags=$(get_desktop_tags | tr - \~ | sort -V -r | tr \~ - | awk '{if(NR>1)print}')
+
+# get tag for previous stable release from the sorted list
+# (first match without `-`, e.g. v1.2.3, not v1.2.3-alpha1)
+last_stable_tag=$(for tag in $tags; do if [[ ! "$tag" == *"-"* ]]; then
+  echo "$tag"
+  break
+fi; done)
+
+# get the current tag, fall back to HEAD
+current_tag=$VERSION
+if [ -z "$current_tag" ]; then
+  current_tag=HEAD
+fi
+
+# Include commit message (%s). Other elements such as
+# commit author (_%aN_) and commit short hash (%h) can
+# be included if desired.
+git_log_format="%s"
+
+echo "## What's Changed"
+echo ""
+
+# Fill and sort changelog (final sort in commit-date order)
+git_log=$(git log --oneline --pretty=format:"$git_log_format" $last_stable_tag...$current_tag -- "$CALYPSO_DIR/desktop/" "$CALYPSO_DIR/client/lib/desktop" |
+  sort -s -k 1,1)
+
+echo "$git_log" | while IFS=$'\r' read change; do
+  awk '$0 !~ /([0-9])+\.([0-9])+\.([0-9])+/ {print "* " $0}' <<< $change
+done
```

Comment:
Nitpick: missing new line at EOF, as per GitHub 🚫 like character in the diff.

The convention of having a new line at the end of the files comes from an old C decision that has been passed down through Unix history:

> A source file that is not empty shall end in a new-line character, which shall not be immediately preceded by a backslash character.
> 
> Since this is a “shall” clause, we must emit a diagnostic message for a violation of this rule.

So, it turns out that, according to POSIX, every text file (including Ruby and JavaScript source files) should end with a `\n`, or “newline” (not “a new line”) character. This acts as the `eol`, or the “end of line” character. It is a line “terminator”. ([Source](https://thoughtbot.com/blog/no-newline-at-end-of-file))

Reply:
ah - didn't know this! Thanks for the education. I added a newline character in 72112d5c6bd4c36d9e406f7c52f3ed687e0ec389.